### ISSUE TITLE
LeaveOneOutEncoder not leaving one out

### ISSUE THREAD

**Issue by discdiver** [scikit-learn-contrib/categorical-encoding#116]
Seems to be just taking the mean of the output for a given level of input.

Similar issue raised previously #38, not clear if OP's  issue was resolved. Code and actual and expected output are below.

No change with
```
randomized = True 
```

Code:
 ```
import numpy as np
import pandas as pd
import category_encoders as ce

df = pd.DataFrame({
    'color':["a", "a", "a", "b", "b", "b"], 
    'outcome':[1,0,0,1,0,1]})

X = df.drop('outcome', axis = 1)
y = df.drop('color', axis = 1)

ce_leave = ce.LeaveOneOutEncoder(cols = ['color'], randomized = True)
ce_leave.fit_transform(X, y['outcome'])

```

Output:

```
	color
0	0.33
1	0.33
2	0.33
3	0.67
4	0.67
5	0.67
```

Expected Output:
```
	color
0	0.00
1	0.50
2	0.50
3	0.50
4	1.00
5	0.50
```

Please advise if I have something wrong.

**Comment by janmotl** [comment]
I think the problem is that the default`fit_transform` method executes:
```python
fit(X, y)
transform(X) # Leave one out cannot be applied because y is not passed
```
instead of:
```python
fit(X, y)
transform(X, y) # Leave one out is applied
```
The pull request is in https://github.com/scikit-learn-contrib/categorical-encoding/pull/110.

In the meantime, do not use `fit_transform` but use directly `fit` and `transform`.

**Comment by discdiver** [comment]
That does it. Thanks @janmotl Thanks for all your work on the pull requests, too!

**Comment by janmotl** [comment]
Solved with #110
